src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtl.cs

```diff
@@ -36,14 +40,43 @@ protected override IEnumerator<RavenEtlItem> ConvertTombstonesEnumerator(IEnumer
 
         protected override EtlTransformer<RavenEtlItem, ICommandData> GetTransformer(DocumentsOperationContext context)
         {
-            return new RavenEtlDocumentTransformer(Database, context, EtlConfiguration);
+            return new RavenEtlDocumentTransformer(Database, context, _script);
         }
 
-        protected override void LoadInternal(IEnumerable<ICommandData> commands, JsonOperationContext context)
+        protected override void LoadInternal(IEnumerable<ICommandData> items, JsonOperationContext context)
         {
-            var batchCommand = new BatchCommand(new DocumentConventions(), context, commands as List<ICommandData>);
+            var commands = items as List<ICommandData>;
+
+            Debug.Assert(commands != null);
+
+            if (commands.Count == 0)
+                return;
+
+            BatchOptions options = null;
+            if (EtlConfiguration.LoadRequestTimeoutInSec != null)
+            {
+                options = new BatchOptions
+                {
+                    RequestTimeout = TimeSpan.FromSeconds(EtlConfiguration.LoadRequestTimeoutInSec.Value)
+                };
+            }
+
+            var batchCommand = new BatchCommand(new DocumentConventions(), context, commands, options);
+            
+            try
+            {
+                _requestExecutor.Execute(batchCommand, context, CancellationToken);
+            }
+            catch (OperationCanceledException e)
+            {
+                if (CancellationToken.IsCancellationRequested == false)
+                {
+                    throw new TimeoutException($"Load request applying the following {commands.Count} commands timed out: " +
+                                               $"{string.Join(", ", commands.Select(x => $"{x.Key} ({x.Method})"))}", e);
```

Comment:
There may be _many_ such items there, maybe better to throw that to the log and just error with the number / size of the request?